### ISSUE TITLE
How to make the ingesting faster?

### ISSUE THREAD

**Issue by smileBeda** [imartinez/privateGPT#152]
I am running the ingesting process on a dataset (PDFs) of 32.2 MB (which I would suppose is not too big?).
The process takes up almost 600% of my CPU

I have a standard Mac, nothing crazy of course.
```
  Model Identifier:	MacBookPro16,1
  Processor Name:	6-Core Intel Core i7
  Processor Speed:	2.6 GHz
  Number of Processors:	1
  Total Number of Cores:	6
  L2 Cache (per Core):	256 KB
  L3 Cache:	12 MB
  Hyper-Threading Technology:	Enabled
  Memory:	16 GB
```

What would be the most required resource of the process to become faster?
Is there any setup that folks recommend here for getting it at least a tad faster?

**Comment by imartinez** [comment]
Ingest got a LOT faster with the use of the new embeddings model #224

Note: this is a breaking change, any existing database will stop working with the new changes. You'll need to re-ingest your docs. It is recommended as the process is faster and the results are better.

**Comment by smileBeda** [comment]
Awesome, trying it out

How do we go about updating?
Just remove the whole thing and start fresh?
Or will a git pull and rerun the requirements.txt?

**Comment by smileBeda** [comment]
Ok it loaded the answer... but it took some 3 minutes at least, for this tiny document.

I fear if I load in my 30MB, this is going to result in a blowup.
Trying it nonetheless now.

**Comment by smileBeda** [comment]
OK, about 10 seconds to ingest for 60KB pdf, about 2 minutes to reply.

Upon entering a followup query, 

```
ggml_new_tensor_impl: not enough space in the context's memory pool (needed 7025376272, available 6997603808)
zsh: segmentation fault  python3 privateGPT.py
bedaschmid@Bedas-MacBook-Pro privateGPT % /usr/local/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
```

I guess _that_ is just a limit of my computer.
